clusterloader2/pkg/measurement/common/service_creation_latency.go

```diff
@@ -0,0 +1,317 @@
+	return []measurement.Summary{summary}, err
+}
+
+func (s *serviceCreationLatencyMeasurement) handleObject(oldObj, newObj interface{}) {
+	var oldService *corev1.Service
+	var newService *corev1.Service
+	var ok bool
+	oldService, ok = oldObj.(*corev1.Service)
+	if oldObj != nil && !ok {
+		klog.Errorf("%s: uncastable old object: %v", s, oldObj)
+		return
+	}
+	newService, ok = newObj.(*corev1.Service)
+	if newObj != nil && !ok {
+		klog.Errorf("%s: uncastable new object: %v", s, newObj)
+		return
+	}
+	isEqual := oldService != nil &&
+		newService != nil &&
+		equality.Semantic.DeepEqual(oldService.Spec, newService.Spec) &&
+		equality.Semantic.DeepEqual(oldService.Status, newService.Status)
+	if isEqual {
+		return
+	}
+
+	if !s.isRunning {
+		return
+	}
+	if newObj == nil {
+		if err := s.deleteObject(oldService); err != nil {
+			klog.Errorf("%s: delete checker error: %v", s, err)
+		}
+		return
+	}
+	if err := s.updateObject(newService); err != nil {
+		klog.Errorf("%s: create checker error: %v", s, err)
+	}
+}
+
+func (s *serviceCreationLatencyMeasurement) deleteObject(svc *corev1.Service) error {
+	key, err := runtimeobjects.CreateMetaNamespaceKey(svc)
+	if err != nil {
+		return fmt.Errorf("meta key created error: %v", err)
+	}
+	s.pingCheckers.DeleteAndStop(key)
+	return nil
+}
+
+func (s *serviceCreationLatencyMeasurement) updateObject(svc *corev1.Service) error {
+	if len(svc.Spec.Ports) < 1 {
+		return nil
+	}
+	key, err := runtimeobjects.CreateMetaNamespaceKey(svc)
+	if err != nil {
+		return fmt.Errorf("meta key created error: %v", err)
+	}
+	if _, exists := s.creationTimes.Get(key, creatingPhase); !exists {
+		s.creationTimes.Set(key, creatingPhase, time.Now())
+	}
+	if len(svc.Status.LoadBalancer.Ingress) < 1 {
+		return nil
+	}
+	if _, exists := s.creationTimes.Get(key, ipAssigningPhase); exists {
+		return nil
+	}
+	s.creationTimes.Set(key, ipAssigningPhase, time.Now())
+	klog.Infof("%s: %s ipAssigned", s, key)
+
+	pc := &pingChecker{
+		callerName:    s.String(),
+		key:           key,
+		svc:           svc,
+		creationTimes: s.creationTimes,
+		stopCh:        make(chan struct{}),
+	}
+	pc.run()
+	s.pingCheckers.Add(key, pc)
+
+	return nil
+}
+
+type pingChecker struct {
+	callerName    string
+	key           string
+	svc           *corev1.Service
+	creationTimes *measurementutil.ObjectTransitionTimes
+	stopCh        chan struct{}
+}
+
+func (p *pingChecker) run() {
+	success := 0
+	for {
+		select {
+		case <-p.stopCh:
+			return
+		default:
+			if _, exists := p.creationTimes.Get(p.key, reachabilityPhase); exists {
+				return
+			}
+			_, err := execservice.RunCommand(
+				fmt.Sprintf("curl %s:%d", p.svc.Status.LoadBalancer.Ingress[0].IP, p.svc.Spec.Ports[0].Port))
+			if err != nil {
+				success = 0
+				time.Sleep(pingBackoff)
+				continue
+			}
+			success++
+			if success == pingChecks {
+				func() {
+					klog.Infof("%v: %s is reachable", p.callerName, p.key)
```

Comment:
I'm not convinced we really would like to log it.

Reply:
Same answer as for logging ipAssigned.